library/spdm_requester_lib/libspdm_req_get_capabilities.c

```diff
@@ -382,6 +423,61 @@ static libspdm_return_t libspdm_try_get_capabilities(libspdm_context_t *spdm_con
         spdm_context->connection_info.capability.max_spdm_msg_size = 0;
     }
 
+    /* Copy algorithms if requested and received */
+    if (get_supported_algorithms && supported_algs != NULL &&
+        spdm_response->header.spdm_version >= SPDM_MESSAGE_VERSION_13 &&
+        (spdm_request->header.param1 & 0x01)) {
+        size_t offset;
+        size_t alg_size;
+        size_t copy_size;
+        size_t i;
+        spdm_supported_algorithms_block_t *src = &spdm_response->supported_algorithms;
+
+        libspdm_responder_supported_algorithms_13_t *dst =
+            (libspdm_responder_supported_algorithms_13_t *)supported_algs;
+        uint8_t *src_bytes = (uint8_t *)src;
+
+        *supported_algs_length = sizeof(libspdm_responder_supported_algorithms_13_t);
+        dst->spdm_version = spdm_response->header.spdm_version;
+        dst->struct_table_count = src->param1;
```

Comment:
I think we need to check if this `param1` exceeds the allowed `struct_table_count`.
Otherwise, it may cause buffer overflow.